native/cocos/core/Root.cpp

```diff
@@ -561,7 +561,9 @@ void Root::doXRFrameMove(int32_t totalFrames) {
     if (_xr->isRenderAllowable()) {
         bool isSceneUpdated = false;
         int viewCount = _xr->getXRConfig(xr::XRConfigKey::VIEW_COUNT).getInt();
-        bool forceUpdateSceneTwice = _xr->getXRConfig(xr::XRConfigKey::EYE_RENDER_JS_CALLBACK).getBool();
+        // compatible native pipeline
+        static bool isNativePipeline = dynamic_cast<cc::render::NativePipeline*>(_pipelineRuntime.get()) != nullptr;
```

Comment:
static might not be needed? one dynamic_cast per-frame should be ok.

Reply:
Reduce dynamic_cast call times as much as possible, there is a performance overhead after all.